btcstaking-tracker/stakingeventwatcher/tracked_delegations.go

```diff
@@ -86,3 +96,28 @@ func (td *TrackedDelegations) RemoveDelegation(stakingTxHash chainhash.Hash) {
 
 	delete(td.mapping, stakingTxHash)
 }
+
+func (td *TrackedDelegations) HasDelegationChanged(
+	stakingTxHash chainhash.Hash,
+	newDelegation *newDelegation,
+) (exists bool, changed bool) {
+	td.mu.Lock()
+	defer td.mu.Unlock()
+
+	// Check if the delegation exists in the map
+	existingDelegation, exists := td.mapping[stakingTxHash]
+	if !exists {
+		// If it doesn't exist, return false for changed, and false for exists
+		return false, false
+	}
+
+	// Compare fields to check if the delegation has changed
+	if existingDelegation.StakingOutputIdx != newDelegation.stakingOutputIdx ||
+		!reflect.DeepEqual(existingDelegation.UnbondingOutput, newDelegation.unbondingOutput) ||
```

Comment:
hmm I think the only thing that could have changes is `.DelegationStartHeight` ? I.e 
- `.StakingOutputIdx` will always be the same
- `.UnbondingOutput` will always be the same
-  `.DelegationStartHeight ` can change from `0` to some other value when providing proof of inclusion

So maybe lets be explicit here at compare only `DelegationStartHeight` ?

Reply:
sg will amend